beacon-chain/cache/k8s_fifo.go

```diff
@@ -0,0 +1,355 @@
+/*
+Copyright 2014 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cache
+
+import (
+	"errors"
+	"sync"
+
+	"k8s.io/client-go/tools/cache"
```

Comment:
If we copied over the things we are importing here, do we need the k8s.io/client-go dependency at all? My brief search indicates this file would be the only import.

Reply:
To be clear, I would recommend only importing things we need. We may not care about the `Queue` interface or the specific errors types.